runtime/vm/JVMImage.cpp

```diff
@@ -74,30 +125,69 @@ JVMImage::freeSubAllocatedMemory(void* address)
 }
 
 void
-JVMImage::reallocateImageMemory(UDATA size)
+JVMImage::readImageFromFile()
 {
-	return;
+    OMRPORT_ACCESS_FROM_OMRPORT(getPortLibrary());
+
+    char imageBuffer[_imageSize];
```

Comment:
If I am in a warm run. I would have to read the size from file before creating imageBuffer. Right now we would probably allocateImage then readImage if cold run, but allocateImage sets _imageSize to default size. We have to change that: not needed now unless you have a solution for this but place a task in Jira for this.